### DISCUSSION TITLE
make IpOpt use Mumps as linear solver

### DISCUSSION THREAD

**Discussion by cjl88** [coin-or/Ipopt#454] [Q&A] [answered]
I compiled IpOpt with mumps as the only solver specified on configure cmd line option. Yet by default, it still use Pardiso  as the solver.  

It looks like this is due to this line:
If you want to compile Ipopt with the Pardiso library that is included in Intel MKL, it is sufficient to ensure that MKL is used for the linear algebra routines (Blas/Lapack), see BLAS and LAPACK.
mentioned at https://coin-or.github.io/Ipopt/INSTALL.html#DOWNLOAD_PARDISO.

I do explicitly specify --with-lapack= for MKL in order to force it to use BLAS from MKL.

I tried to compile without pardiso using --without-pardiso, but I got errors from "confiugure":
checking for function pardiso_ipopt_newinterface_ in yes -L/nfs/software/intel/oneapi/mkl/2021.1.1/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_gnu_thread -lmkl_core -lgomp -lpthread -lm -ldl ... no
checking for function pardiso_ipopt_newinterface in yes -L/nfs/software/intel/oneapi/mkl/2021.1.1/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_gnu_thread -lmkl_core -lgomp -lpthread -lm -ldl ... no
checking for function PARDISO_IPOPT_NEWINTERFACE_ in yes -L/nfs/software/intel/oneapi/mkl/2021.1.1/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_gnu_thread -lmkl_core -lgomp -lpthread -lm -ldl ... no
checking for function PARDISO_IPOPT_NEWINTERFACE in yes -L/nfs/software/intel/oneapi/mkl/2021.1.1/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_gnu_thread -lmkl_core -lgomp -lpthread -lm -ldl ... no
configure: error: Symbol pardiso_ipopt_newinterface not found with Pardiso flags yes and Lapack. Require Pardiso >= 4.0.

I know it's using pardiso because this is output at runtime:
This is Ipopt version 3.13.3, running with linear solver pardiso.

So can I make IpOpt to use Mumps at runtime?

Thanks.

**Comment by cjl88** [comment]
Found a programmatic way to use mumps at run time.

**Comment by svigerske** [comment, marked as answer]
Yes, there is no way to disable the check in configure whether Pardiso is included with Blas/Lapack (in which case it will assume that it is MKL Pardiso). d61276c4 changes this to allow to use `--without-pardiso` to disable the checks for Pardiso. This should be in Ipopt 3.13.5.

PS: `--without-pardiso` was changed to `--disable-pardisomkl` in Ipopt 3.14.
